src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -45,65 +43,74 @@ ArgumentSyntax syntax
 			context.ConfigureGeneratedCodeAnalysis( GeneratedCodeAnalysisFlags.None );
 
-			context.RegisterSyntaxNodeAction(
-				AnalyzeCallSyntax,
-				SyntaxKind.InvocationExpression
-			);
+			context.RegisterCompilationStartAction( OnCompilationStart );
+		}
 
-			context.RegisterSyntaxNodeAction(
-				AnalyzeCallSyntax,
-				SyntaxKind.ObjectCreationExpression
-			);
+		private static void OnCompilationStart( CompilationStartAnalysisContext context ) {
+			INamedTypeSymbol? requiredNamedArguments = context.Compilation.GetTypeByMetadataName( RequireNamedArgumentsAttribute );
+			if( requiredNamedArguments is null ) {
+				return;
+			}
 
-			context.RegisterSyntaxNodeAction(
-				AnalyzeCallSyntax,
-				SyntaxKind.ThisConstructorInitializer
-			);
+			INamedTypeSymbol? lambdaExpresssion = context.Compilation.GetTypeByMetadataName( LambdaExpressionMetadataName );
 
-			context.RegisterSyntaxNodeAction(
-				AnalyzeCallSyntax,
-				SyntaxKind.BaseConstructorInitializer
+			context.RegisterOperationAction(
+				ctx => AnalyzeInvocation(
+					ctx,
+					requiredNamedArguments,
+					lambdaExpresssion
+				),
+				OperationKind.Invocation,
+				OperationKind.ObjectCreation
 			);
 		}
 
-		private static void AnalyzeCallSyntax(
-			SyntaxNodeAnalysisContext ctx
+		private static void AnalyzeInvocation(
+			OperationAnalysisContext ctx,
+			INamedTypeSymbol requireNamedArgumentsSymbol,
+			INamedTypeSymbol? lambdaExpresssionSymbol
 		) {
+			(IMethodSymbol targetMethod, ImmutableArray<IArgumentOperation> args) = ctx.Operation switch {
+				IInvocationOperation op => (op.TargetMethod, op.Arguments),
+				IObjectCreationOperation op => (op.Constructor!, op.Arguments),
+				_ => default
+			};
 
-			ArgumentListSyntax args = GetArgs( ctx.Node );
-			if( args == null ) {
+			if( targetMethod == default || args == default ) {
 				return;
 			}
 
+			if( args.IsEmpty ) {
+				return;
+			}
+
+			bool requireNamedArguments = HasRequireNamedArgumentsAttribute( requireNamedArgumentsSymbol, targetMethod );
+
 			// Don't complain about single argument functions because they're
 			// very likely to be understandable
-			if( args.Arguments.Count <= 1 ) {
+			if( !requireNamedArguments && args.Length == 1 ) {
 				return;
 			}
 
-			// Don't complain about expression trees, since they aren't allowed
-			// to have named arguments
-			if( IsExpressionTree( ctx.Node, ctx.SemanticModel ) ) {
+			ImmutableArray<ArgParamBinding> unnamedArgs = GetUnnamedArgs( args ).ToImmutableArray();
+			if( unnamedArgs.IsEmpty ) {
 				return;
 			}
 
-			ImmutableArray<ArgParamBinding> unnamedArgs =
-				GetUnnamedArgs( ctx.SemanticModel, args, ctx.CancellationToken )
-				.ToImmutableArray();
-
-			if( unnamedArgs.IsEmpty ) {
+			// Don't complain about expression trees, since they aren't allowed
+			// to have named arguments
+			if( IsExpressionTree( lambdaExpresssionSymbol, ctx.Operation.SemanticModel!, ctx.Operation ) ) {
 				return;
 			}
```

Comment:
I didn't try to optimize this, and I expect its still the most expensive part. It can probably be updated to stay in operation land rather than syntax.